### PR TITLE
Bump [compat] FilePathsBase="0.6,0.7,0.8,0.9"

### PR DESCRIPTION
**Pull request by ablaom**
We need this to eliminate package conflicts at test time in MLJBase and related packages.

### CODE REVIEW

File: test/write.jl [hunk 1]
```diff
@@ -1,4 +1,6 @@
 using CSV, Dates, WeakRefStrings, CategoricalArrays, Tables
+using FilePathsBase
```
Comment (quinnj) [inline test/write.jl:2]: Why is this needed?

Comment (ablaom) [reply test/write.jl:2]: Possibly you only need the line after: `using FilePathsBase: /`. Without that the FilePaths test fail.

Comment (quinnj) [reply test/write.jl:2]: @oxinabox @rofinn , do either of you know why this would now be needed? It seems odd to me. Did FilePathsBase stop exporting something or something?

Comment (rofinn) [reply test/write.jl:2]: Yes, `/` and `join` no longer extend the base methods as requested by some core dev team members. If you want those methods to work with a path type then you need to do `using FilePathsBase: /, join`. FWIW, we might be able to think of a way for FilePathsBase to be a test-only dependency if we loosen a couple of type restrictions?

**Comment by ExpandingMan** [comment]
Didn't notice this PR when I wrote #636.  This issue is really wreaking havoc for me.

I notice this is lacking tests for 1.4.  Could we add those and then merge this?

**Comment by rofinn** [comment]
I'm fine with merging this as is if Jacob Quinn doesn't have any objections in the next day or so. I think we should probably try to drop the primary FilePathsBase dependency and loosen the type restrictions, but that can happen later.

**Comment by oxinabox** [comment]
We had that before with no dependency and it working.
Then FilePathBase changed something and it broke.
And the package manager didn't know since there was no dependency.

Then when the fix was made a dependency was added.

**Comment by rofinn** [comment]
I’m guessing that was when AbstractPath stopped subtyping AbstractString, as this package now just converts to a string. There’s probably a better way to handle that and we should still continue testing against FilePathsBase.

**Comment by oxinabox** [comment]
> I’m guessing that was when AbstractPath stopped subtyping AbstractString, a

No it was after that because 
https://github.com/JuliaData/CSV.jl/pull/405 was made when FilePaths stopped subtyping String (0.5)

where as the PR that added the dependency was:
https://github.com/JuliaData/CSV.jl/pull/511/

**Comment by ablaom** [comment]
FYI The only CI fail is a time-out on Windows julia 1.0

```
curl: (7) Failed to connect to storage.googleapis.com port 443: Timed out
```
